### ISSUE TITLE
Parser fails on {#snippet} with typed destructuring and trailing comma

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#57]
## Great progress in v0.5.6! 🎉

First, thank you for the rapid fixes! Testing against our codebase ([matterviz](https://github.com/janosh/matterviz)):

| Version | Errors | Warnings |
|---------|--------|----------|
| v0.5.4 | 154 | 22 |
| v0.5.5 | 46 | 22 |
| **v0.5.6** | **4** | 22 |

That's a **97% reduction** in errors from v0.5.4 → v0.5.6!

## Remaining Issue

`{#snippet}` with typed destructuring parameters and a trailing comma fails to parse:

```svelte
{#snippet user_content(
  { x_scale, width, height }: UserContentProps,
)}
  {@const x0 = x_scale(0)}
  <line x1={x0} ... />
{/snippet}
```

Error: `Variable declaration expected. (ts(TS1134))`

## Reproduction

https://github.com/janosh/svelte-check-rs-repro

See `src/lib/SnippetWithTypedDestructuring.svelte`

```bash
git clone https://github.com/janosh/svelte-check-rs-repro
cd svelte-check-rs-repro
pnpm install
node node_modules/svelte-check-rs/install.js
pnpm check     # svelte-check passes ✅
pnpm check:rs  # svelte-check-rs fails ❌
```

## Analysis

The trailing comma after the type annotation (`}: UserContentProps,`) seems to trigger the error. Without the trailing comma it may work, but Svelte/prettier often adds trailing commas.

## Environment

- svelte-check-rs: 0.5.6
- svelte-check: 4.3.5 (passes on same code)
- svelte: 5.46.1
